src/wallet/rpcwallet.cpp

```diff
@@ -3766,14 +3766,7 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
     ScriptPubKeyMan* spk_man = pwallet->GetScriptPubKeyMan();
     if (spk_man) {
         CKeyID key_id = GetKeyForDestination(*provider, dest);
```

Comment:
Maybe add a comment or assertion that only LegacyScriptPubKeyMan should use `key_id`.